src/app/page/MakeFoodBoxMission/MissionForm.js

```diff
@@ -0,0 +1,246 @@
+      func = setDropOff;
+      rest = dropOff;
+    }
+    if (stage === "pickUp") {
+      func = setPickUp;
+      rest = pickUp;
+    }
+    if (query.suggestion) {
+      const { value, latlng, county, countryCode } = query.suggestion;
+      func({
+        ...rest,
+        location: {
+          address: value,
+          geoLocation: latlng,
+          county,
+          countryCode,
+        },
+      });
+    }
+  };
+
+  const handleDate = (date, stage) => {
+    if (!date) {
+      return;
+    }
+    if (stage === "pickUp") {
+      setPickUpDateLabel(date);
+      if (typeof date !== "string") {
+        setPickUp({ ...pickUp, date: date.toString().substr(0, 15) });
+      } else {
+        setPickUp({ ...dropOff, date: date || null });
+      }
+    }
+    if (stage === "dropOff") {
+      setDropOffDateLabel(date);
+      if (typeof date !== "string") {
+        setDropOff({ ...dropOff, date: date.toString().substr(0, 15) });
+      } else {
+        setDropOff({ ...dropOff, date: date || null });
+      }
+    }
+  };
+
+  return (
+    <Page template="pink">
+      <Container classes={{ root: classes.root }}>
+        <StyledHeader main align="center" variant="h1">
+          Create a food box mission.
+        </StyledHeader>
+        <StyledHeader align="left" variant="h2">
+          Recipient
+        </StyledHeader>
+        <TextField
+          fullWidth={true}
+          variant="outlined"
+          value={values.recipient || ""}
+          name="recipient"
+          onChange={handleChange}
+          label="Assign a recipient"
+          helperText="Handle or phone number"
+        />
+        <StyledHeader align="left" variant="h2">
+          Helper
+        </StyledHeader>
+        <TextField
+          fullWidth={true}
+          variant="outlined"
+          value={values.helper || ""}
+          name="helper"
+          onChange={handleChange}
+          label="Assign a helper"
+          helperText="Handle or phone number"
+          //disabled={autoAssigned}
+        />
+        {/*<FormControlLabel
+          control={<Checkbox name="autoAssignHelper" onChange={assignHelper} />}
+          label="Auto-assign a helper"
+        />*/}
+        <StyledHeader align="left" variant="h2">
+          Additional info
+        </StyledHeader>
+        <TextField
+          fullWidth={true}
+          variant="outlined"
+          value={values.description || ""}
+          name="description"
+          multiline={true}
+          rows={4}
+          onChange={handleChange}
+          label="Mission notes"
+          required={true}
+        />
+        <MuiPickersUtilsProvider utils={MomentUtils}>
+          <StyledHeader align="left" variant="h2">
+            Pickup
+          </StyledHeader>
+          <AlgoliaPlaces
+            placeholder="Pickup location"
+            name="pickUp"
+            options={{
+              appId: "plZ318O8ODTC",
+              apiKey: "b5e0781d289a9aa8edb37bf24aef874e",
```

Comment:
We should break these out into consts and share between the components

Reply:
Thanks a lot for reviewing. I'll keep a single [PR](https://github.com/factn/resilience-app/pull/211) , and there I shall correct what you guys have pointed. I think FoodBox mission can be just a type to be chosen, since it's part of the same collection, we don't need a separate route. @mat10tng @utunga thoughts?